src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -833,6 +875,12 @@ RangesInDataParts MergeTreeDataSelectExecutor::filterPartsByPrimaryKeyAndSkipInd
                 auto & stat = merged_indices_stat[idx];
                 stat.total_parts.fetch_add(1, std::memory_order_relaxed);
 
+                if (auto result = can_use_merged_index(indices_and_condition.indices); !result)
```

Comment:
Just for my understanding, if any one of the merged indices cannot be used as a skip index, we continue (i.e not use it for filtering) right ?

Reply:
Yes